src/app/features/worklog/worklog-export/worklog-export.component.ts

```diff
@@ -350,6 +384,16 @@ export class WorklogExportComponent implements OnInit, OnDestroy {
         group.notes[i] = group.notes[i].replace(/\n/g, ' - ');
       }
 
+      group.projects = unique(group.projects).filter((project: string) => !!project);
```

Comment:
Both unique and filter should not be necessary here and cost performance. If a project is undefined than we got a data error which we want to see.

Reply:
I agree, but I would say there is more to that.
1. just removing those 3 lines (including the one for notes) results in errors and wrong (empty) outputs.
2. to avoid that, we'd have to:
 - use the ternary operator in several places throughout lines ~270-326 and 
 - push only if not already exists in lines ~337-341
3. to begin with, I did this to change as little code as possible and keep its structure as it is: first we create the rows and just push whatever we get from the list of tasks and _only after_ we reiterate and clean out every single row.
5. so since we are talking about refactoring we could move this to a new issue/PR?
6. another thing is that we do that for notes/tags/projects even if they're not even part of the columns to show...

If you want I can also commit changes as explained in point 2. and you chose what you like better :)
And don't get me wrong I totally agree on what you said.

Reply:
Yes, you're right! Sorry. I slightly misread the part to be honest. What confused me was that at different points of time `group.project` and `group.tag` mean different things, which really is äh confusing! :)  But I also saw just now, that this is just consistent with the other stuff being done here and IMHO you're definitely right about prioritizing consistency over just doing it sometimes right! Let's move this to a new PR!